### PR TITLE
Add support for project-wide occurrences

### PR DESCRIPTION
**Pull request by voodoos**

### CODE REVIEW

File: src/analysis/occurrences.ml [hunk 1]
```diff
@@ -87,12 +95,13 @@ let index_buffer =
...
-  fun ~current_buffer_path ~stamp ~local_defs () ->
+  fun ~scope ~current_buffer_path ~stamp ~local_defs () ->
     let {Logger. log} = Logger.for_section "index" in
     match !cache with
-    | Some (path, stamp', value) when
+    | Some (path, stamp', scope', value) when
         String.equal path current_buffer_path
-        && Int.equal stamp' stamp ->
+        && Int.equal stamp' stamp
+        && scope' = scope ->
```
Comment (xvw) [inline src/analysis/occurrences.ml:104]: Could we benefit to add a `Scope.equal` (probably written as `let equal = (=)` in order to try to get rid of `=` and providing, in a future, an implementation of scopes equality?

File: src/index-format/dune [hunk 2]
```diff
@@ -0,0 +1,10 @@
+(library
+ (name merlin_index_format)
```
Comment (xvw) [inline src/index-format/dune:2]: Why the name `index_format`?

**Review by xvw — APPROVED** [review]
👍🏾

File: src/index-format/dune (hunk 2 above)
Comment (voodoos) [reply src/index-format/dune:2]: This was answered offline: it uses the same convention as compiler artifacts: `cmt_format`, `cmi_format` etc.

**Pull request merged**